contracts/SafeOwn.sol

```diff
@@ -0,0 +1,105 @@
+pragma solidity ^0.8.17;
+
+/**
+* @dev Contract defines way for 2-setp access control for the owner of contract.
+* To avoid accidental transfer owner will have to first propose a new owner
+* then new owner have to accept the ownership.
+*/
+abstract contract SafeOwn {
+	address private _Owner;
+	address private _pendingOwner;
+
+	/**
+	* @dev Emitted when the Ownership is transferred.
+     */
+	event ownershipTransferred(
```

Comment:
Event name Case is not consistent with PaymentVault.sol contract.
Check other events also